src/shared/extensions/yamlActivation.ts

```diff
@@ -0,0 +1,207 @@
+        const openTemplateYamls = vscode.window.visibleTextEditors.filter(editor => {
+            const fileName = editor.document.fileName
+            return getAWSFileName(fileName) !== undefined
+        })
+
+        if (openTemplateYamls.length > 0) {
+            promptInstallYamlPluginFromEditor(openTemplateYamls[0], yamlPromptDisposables)
+        }
+    }
+}
+
+/**
+ * Gets the matching aws file type from a filename
+ * @param fileName The name of the incoming file
+ * @returns The matching aws file type or undefined
+ */
+function getAWSFileName(fileName: string): AWSFiletypeNames | undefined {
+    for (const [type, names] of Object.entries(awsFiletypeFileNames)) {
+        for (const name of names) {
+            if (fileName.endsWith(name)) {
+                return type as AWSFiletypeNames
+            }
+        }
+    }
+    return undefined
+}
+
+async function promptInstallYamlPluginFromEditor(
+    editor: vscode.TextEditor | undefined,
+    disposables: vscode.Disposable[]
+): Promise<void> {
+    if (editor) {
+        promptInstallYamlPluginFromFilename(editor.document.fileName, disposables)
+
+        if (!path.isAbsolute(editor.document.uri.fsPath)) {
+            return
+        }
+
+        if (globals.templateRegistry.cfn?.getRegisteredItem(editor.document.uri)) {
+            promptInstallYamlPlugin(disposables, 'CloudFormation')
+        }
+
+        if (globals.templateRegistry.buildspec?.getRegisteredItem(editor.document.uri)) {
+            promptInstallYamlPlugin(disposables, 'Buildspec')
+        }
+    }
+}
+
+/**
+ * Prompt user to install YAML plugin for template.yaml and template.yml files
+ * @param fileName File name to check against
+ * @param disposables List of disposables to dispose of when the filename is a template YAML file
+ */
+async function promptInstallYamlPluginFromFilename(fileName: string, disposables: vscode.Disposable[]): Promise<void> {
+    const awsFileName = getAWSFileName(fileName)
+    if (awsFileName) {
+        promptInstallYamlPlugin(disposables, awsFileName)
+    }
+}
+
+/**
+ * Show the install YAML extension prompt and dispose other listeners
+ * @param disposables
+ */
+async function promptInstallYamlPlugin(disposables: vscode.Disposable[], templateName?: AWSFiletypeNames) {
+    // immediately dispose other triggers so it doesn't flash again
+    for (const prompt of disposables) {
+        prompt.dispose()
+    }
+    const settings = PromptSettings.instance
+
+    const installBtn = localize('AWS.missingExtension.install', 'Install...')
+    const permanentlySuppress = localize('AWS.message.info.yaml.suppressPrompt', "Don't show again")
+
+    const template = templateName ? templateName : 'YAML'
+    const response = await vscode.window.showInformationMessage(
+        localize(
+            'AWS.message.info.yaml.prompt',
+            'Install YAML extension for more {0} features in {1} templates',
```

Comment:
This would read "devfile templates" and "buildspec templates". Maybe we should drop "templates" from the message or change it to "files"?

```suggestion
            'Install YAML extension for more {0} features in {1} files',
```